### ISSUE TITLE
[NEW] Неверное изменение переменной\реквизита внутри цикла - имеет значение только последняя итерация цикла

### ISSUE THREAD

**Issue by artbear** [1c-syntax/bsl-language-server#1522]
## Описание проблемы, ошибки, которую надо диагностировать

Разработчики иногда ошибаются, выполняя внутри цикла установку одного и того же значения\реквизита.
В таком случае фактически последняя запись перезапишет все остальные, т.е. предыдущие итерации не имеют смысла. 
Часто подобные ошибки возникают из копи-паста

Неверно
```bsl
Для Каждого ЭлементЦеныПоставки Из ЦеныПоставок.НайтиСтроки(КлючПоискаЦенПоставок) Цикл
  ОписаниеПоставки.Цена= ЭлементЦеныПоставки.Цена;
  ОписаниеПоставки.Валюта = XMLСтрока(ЭлементЦеныПоставки.ВалютаСсылка);
КонецЦикла;
```
или
```bsl
Для Каждого Элемент Из Коллекция Цикл
  Значение = Элемент.Цена;
КонецЦикла;
```

Нужно учесть
- обычную переменную - Значение =
- реквизита объекта - Объект.Реквизит = 
  -  с разным уровнем вложенности - Объект.Реквизит...ВложенныйРеквизит = 

### Тип [Статья на русском](https://1c-syntax.github.io/bsl-language-server/contributing/DiagnosticTypeAndSeverity)

* [ ] :ant: Ошибка
* [ ] :cop: Уязвимость
* [ ] :guardsman: Потенциальная уязвимость
* [x] :poop: Качество кода
* [ ] :trollface: Другое

### Важность [Статья на русском](https://1c-syntax.github.io/bsl-language-server/contributing/DiagnosticTypeAndSeverity)

* [ ] :broken_heart: Блокирующая / Blocker
* [ ] :heart: Критическая / Critical
* [x] :yellow_heart: Важная / Major
* [ ] :blue_heart: Незначительная / Minor
* [ ] :green_heart: Информационная / Info
* [ ] :revolving_hearts: Другое

### Тэги [Статья на русском](https://1c-syntax.github.io/bsl-language-server/contributing/DiagnosticTag)

* [ ] `STANDARD` - "Нарушение стандартов 1С"
* [ ] `LOCKINOS` - "Не будет работать в другой ОС"
* [ ] `SQL` - "Проблема с запросом"
* [ ] `PERFORMANCE` - "Проблема производительности"
* [ ] `BRAINOVERLOAD` - "Непонятный код"
* [x] `BADPRACTICE` - "Плохая практика программирования"
* [ ] `CLUMSY` - "Излишние действия"
* [ ] `DESIGN` - "Ошибка в проектировании"
* [x] `SUSPICIOUS` - "Подозрительный код"
* [x] `UNPREDICTABLE` - "Непредсказуемо работающий код"
* [ ] `DEPRECATED` - "Устаревшая функциональность"
* [ ] `ERROR` - "Ошибочная конструкция"
* [ ] `LOCALIZE` - "Проблемы локализации"

### Время на исправление (минут)

3 минуты

**Comment by artbear** [comment]
Фактически это отдельная диагностика из разряда затираемых\скрываемых переменных из #1088

**Comment by nixel2007** [comment]
Я бы не стал утверждать, что это однозначно ошибка. Частый кейс - поиск последнего значения в цикле по коллекции с какими-то условиями/отборами. Возможно можно считать, что если узел присваивания в переменной находится в такой точке графа потока управления, через которую гарантировано проходят все итерации (например, напрямую в теле цикла, без вложенных условий, или в условиях, которые всегда выполняются, но тут без DFA никак), то это подозрительное место, но надо еще подумать.

**Comment by artbear** [comment]
Да, если есть условие внутри, тут правило нужно аккуратно проверить, возможно, исключить для циклов с условиями.

но я несколько раз нарывался, что такая ошибка в цикле без условий встречается.

пример кода из реального коммита (

кажется, я что-то подобное в правилах из открытых плагинов Сонара встречал.

**Comment by artbear** [comment]
Точно уже не помню, возможно, я такое правило реализовывал в Пульском плагине.

**Comment by zeegin** [comment]
Кажется будет много FP.
Я бы не стал делать.
Я не считаю это подозрительным кодом.

Может быть много случаев выхода из цикла до или после присваивания.
